pkgs/development/libraries/lcrq/default.nix

```diff
@@ -0,0 +1,28 @@
+{
+  stdenv,
+  fetchFromGitea,
+  lib
+}:
+stdenv.mkDerivation (finalAttrs: {
+  name = "lcrq";
+  version = "0.1.0";
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "librecast";
+    repo = "lcrq";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-s8+uTF6GQ76wG1zoAxqCaVT1J5Rd7vxPKX4zbQx6ro4=";
+  };
+
+  installFlags = [ "PREFIX=$(out)" ];
+
+  meta = {
+    changelog = "https://codeberg.org/librecast/lcrq/src/tag/v${finalAttrs.version}/CHANGELOG.md";
+    description = "Librecast RaptorQ library.";
+    homepage = "https://librecast.net/lcrq.html";
+    license = [ lib.licenses.gpl2 lib.licenses.gpl3 ];
+    maintainers = with lib.maintainers; [ albertchae aynish DMills27 jasonodoom jleightcap ];
+    platforms = lib.platforms.gnu;
```

Comment:
We originally had `platforms.unix` but we ran into a compilation issue with `gcc` vs `clang` on darwin. Before this change we receive the below error:

```
> ld: unknown option: -soname
```

According to [this issue](https://github.com/CopernicaMarketingSoftware/PHP-CPP/issues/368#issuecomment-367825982) this is a known difference between clang+llvm and gcc. Is there a workaround or is it more appropriate to report to the project authors?

Reply:
Dear @NixOS/darwin-maintainers ,

Can you help with this comment?

Thanks

Reply:
This should definitely go upstream, but maybe try

```nix
postPatch = lib.optionalString stdenv.isDarwin ''
  substituteInPlace src/Makefile.in \
    --replace .so .dylib \
    --replace -soname -install_name
'';
```